### DISCUSSION TITLE
Lazyloading

### DISCUSSION THREAD

**Discussion by arthurhamon** [dromru/react-photoswipe-gallery#1135] [Q&A]
Hey,

Is their an option to do lazyloading on the image gallery?
My component will show the first image, then I want the second image to be loaded only when the gallery is open and user ask for second image.

Is this something already available or doable?

Thanks,
Arthur

**Comment by akhmadullin** [comment]
Hi

What image do you mean?

Do you mean thumbnails

<img width="1665" alt="Снимок экрана 2023-04-21 в 13 38 35" src="https://user-images.githubusercontent.com/37641303/233535545-20c5e0cf-694d-4264-964d-f902e3922b14.png">

or

image inside opened Photoswipe

<img width="1665" alt="Снимок экрана 2023-04-21 в 13 38 39" src="https://user-images.githubusercontent.com/37641303/233535615-db7bfe5e-52b9-4d46-b1b3-abba7b65b433.png">

?

**Comment by arthurhamon** [comment]
Thank you, I talking about image inside opened Photoswipe.

**Comment by akhmadullin** [comment]
It seams, that preloading of neighboring slides can't be disabled - https://photoswipe.com/options/#preload

Photoswipe always loads two nearby images, you can only increase number of slides to be preloaded, but can't disable this feature

**Comment by DeFuex** [comment]
What I did is creating a custom LazyImage component that loads when the image is in the users viewport. And use this in the content property of <Item> instead of using original property with the image url. Not sure if this should be done, but it seems to work so far for me. In case of images seen i locally cash them in some "map" so that there's no reloading triggered when used in viewport mode. You could potentially use this component also for the thumbnails in the gallery.
